client/rest-high-level/src/main/java/org/elasticsearch/client/AsyncSearchRequestConverters.java

```diff
@@ -73,7 +73,9 @@ static void addSearchRequestParams(Params params, SubmitAsyncSearchRequest reque
         if (request.getAllowPartialSearchResults() != null) {
             params.withAllowPartialResults(request.getAllowPartialSearchResults());
         }
-        params.withBatchedReduceSize(request.getBatchedReduceSize());
+        if (request.getBatchedReduceSize() != SubmitAsyncSearchRequest.DEFAULT_BATCHED_REDUCE_SIZE) {
```

Comment:
I would prefer if we found a way to not have the default value in the client too. Would it be an option to make this an Integer that is null by default, and only set it when it was explicitly set?

Reply:
That would be an Integer in the SubmitAsyncSearchRequest, masking the one in the wrapped SearchRequest, I think. I find that equally confusing from the code side tbh, we would keep most parameters set on the internal search request but only this in the async submit request? With the current defautl its at least clear when looking at the code what value is used when the user doesn't set this? wdyt?